src/types.ts

```diff
@@ -0,0 +1,115 @@
+import type { Region } from '@saucelabs/testcomposer';
+
+export type Browser = 'chromium' | 'firefox' | 'webkit' | 'chrome';
+
+export interface Metrics {
+  name: string;
+  data: {
+    install: {duration: number};
+    rebuild?: {duration: number};
+    setup: {duration: number};
+  };
+}
+
+export interface RunResult {
+  startTime: string;
+  hasPassed: boolean;
+  endTime: string;
+  metrics: Metrics[];
+}
+
+export interface RunnerConfig {
+  // NOTE: Kind is serialized by saucectl with a capital 'K' ¯\_(ツ)_/¯
```

Comment:
😢 

It's because it's missing JSON tags:
```type TypeDef struct {
	APIVersion string `yaml:"apiVersion,omitempty"`
	Kind       string `yaml:"kind,omitempty"`
}

The default serialization in Go will serialize the field name as-is. Since public fields start with a capital letter, the serialized value will be the same 😅

Reply:
👍 yah, I know. Just wanted to note it for future reference.